packages/next/server/static-paths-worker.ts

```diff
@@ -0,0 +1,40 @@
+import { join } from 'path'
+import { buildStaticPaths } from '../build/utils'
+import { getPagePath } from '../next-server/server/require'
+import { loadComponents } from '../next-server/server/load-components'
+import { PAGES_MANIFEST, SERVER_DIRECTORY } from '../next-server/lib/constants'
+
+// we call getStaticPaths in a separate process to ensure
+// side-effects aren't relied on in dev that will break
+// during a production build
+export async function loadStaticPaths(
+  distDir: string,
+  buildId: string,
+  pathname: string,
+  serverless: boolean
+) {
+  // we need to clear any modules manually here since the
+  // require-cache-hot-loader doesn't affect require cache here
+  // since we're in a separate process
+  delete require.cache[join(distDir, SERVER_DIRECTORY, PAGES_MANIFEST)]
```

Comment:
What happens if we `process.exit(0)` after we're done? Does Jest restart the worker and make a new clean one?

Reply:
This would be expensive since we would be incurring the worker boot-up each time we called `getStaticPaths`. We might be able to  call `setImmediate(() => process.exit(1))` and then return the static paths hoping they are sent back to the parent before the process is killed but this seems dangerous and more efficient to clear the cache like we do in the main process and re-use the same worker for calling `getStaticPaths`